src/components/FindMovie/FindMovie.js

```diff
@@ -1,55 +1,113 @@
-import React from 'react';
-import './FindMovie.scss';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
 
+import { getFilm } from '../../api/api';
 import { MovieCard } from '../MovieCard';
-import movies from '../../api/movies.json';
-
-export const FindMovie = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+import './FindMovie.scss';
+
+export const FindMovie = ({ addMovie }) => {
+  const [searchQuery, setsearchQuery] = useState('');
+  const [newMovie, setNewMovie] = useState(null);
+  const [searchQueryError, setSearchQueryError] = useState(false);
+
+  const searchMovie = () => {
+    if (!searchQuery) {
+      return;
+    }
+
+    getFilm(searchQuery).then((result) => {
+      if (result.Error) {
+        setSearchQueryError(true);
+
+        return;
+      }
+
+      setNewMovie({
+        title: result.Title,
+        description: result.Plot,
+        imgUrl: result.Poster,
+        imdbId: result.imdbID,
+      });
+      setSearchQueryError(false);
+    });
+  };
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+  const onAddMovie = () => {
+    addMovie((movies) => {
+      if (newMovie === null) {
+        return movies;
+      }
+
+      if (movies.find(movie => movie.title === newMovie.title) !== undefined) {
+        return movies;
+      }
+
+      setNewMovie(null);
+      setsearchQuery('');
+
+      return [...movies, newMovie];
+    });
+  };
+
+  return (
+    <>
+      <form className="find-movie">
+        <div className="field">
+          <label className="label" htmlFor="movie-title">
+            Movie title
+          </label>
+
+          <div className="control">
+            <input
+              type="text"
+              id="movie-title"
+              placeholder="Enter a title to search"
+              className={classNames('input', { 'is-danger': searchQueryError })}
+              onChange={event => setsearchQuery(event.target.value)}
```

Comment:
Also you can reset an error here
`setSearchQueryError(false)` to hide the error when the user starts to enter new query